frontend/README.md

```diff
@@ -1,36 +1,207 @@
-This is a [Next.js](https://nextjs.org) project bootstrapped with [`create-next-app`](https://nextjs.org/docs/app/api-reference/cli/create-next-app).
+# NodeGaze Frontend 🎯
 
-## Getting Started
+The modern web interface for NodeGaze, built with Next.js 14, React, and TypeScript. Provides a comprehensive dashboard for monitoring Lightning Network nodes with real-time event tracking and notification management.
 
-First, run the development server:
+## 🚀 Features
+
+### Dashboard & Monitoring
+
+- **Real-time Event Streaming**: Live updates of Lightning Network events
+- **Event Filtering**: Filter by event type (Invoice, Channel operations) and severity
+- **Event History**: Paginated event lists with detailed information
+- **Node Information**: Display node aliases, IDs, and connection status
+
+### Notification Management
+
+- **Webhook Configuration**: Set up HTTP endpoints for event notifications
+- **Discord Integration**: Send formatted alerts to Discord channels
+- **Event Type Selection**: Choose which events to forward to each notification endpoint
+- **Notification Status**: Track delivery status and retry failed notifications
+
+### User Experience
+
+- **Responsive Design**: Works seamlessly on desktop, tablet, and mobile
+- **Modern UI**: Clean interface built with shadcn/ui components
+- **Authentication**: Secure login/signup with NextAuth.js integration
+- **Dark/Light Theme**: Support for user preference themes (coming soon)
+
+## 🛠️ Tech Stack
+
+- **Framework**: Next.js 14 (App Router)
+- **Language**: TypeScript
+- **Styling**: Tailwind CSS
+- **UI Components**: shadcn/ui
+- **Authentication**: NextAuth.js
+- **State Management**: React Hooks + Context
+- **HTTP Client**: Native fetch API
+- **Icons**: Lucide React
+
+## 🚀 Getting Started
+
+### Prerequisites
+
+- Node.js 18+
+- npm, yarn, or pnpm
+- NodeGaze backend running on `http://localhost:3030`
+
+### Installation
+
+1. **Install dependencies**
+
+   ```bash
+   npm install
+   # or
+   yarn install
+   # or
+   pnpm install
+   ```
+
+2. **Environment Setup**
+
+   ```bash
+   cp .env.example .env.local
+   ```
+
+   Configure your environment variables:
+
+   ```bash
+   # API Configuration
+   BACKEND_URL=http://localhost:3030
+   NEXTAUTH_URL=http://localhost:3000
+   NEXTAUTH_SECRET=your-nextauth-secret-key
+   
+   # JWT Configuration
+   NEXTAUTH_JWT_SECRET=your-jwt-secret-key
+   ```
+
+3. **Start Development Server**
+
+   ```bash
+   npm run dev
+   # or
+   yarn dev
+   # or
+   pnpm dev
+   ```
+
+4. **Open in Browser**
+   Navigate to [http://localhost:3000](http://localhost:3000)
+
+## 📁 Project Structure
```

Comment:
do we need to include project structure,

Reply:
done